exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -526,6 +534,48 @@ private void writeFile(ParquetTableMetadataDirs parquetTableMetadataDirs, Path p
   }
 
   /**
+   * Serializer for ParquetPath. Writes the path relative to the root path
+   */
+  private static class ParquetPathSerializer extends StdSerializer<ParquetPath> {
+    private final String rootPath;
+
+    ParquetPathSerializer(String rootPath) {
+      super(ParquetPath.class);
+      this.rootPath = rootPath;
+    }
+
+    @Override
+    public void serialize(ParquetPath parquetPath, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException, JsonGenerationException {
+      Preconditions.checkState(parquetPath.getFullPath().startsWith(rootPath), String.format("Path %s is not a subpath of %s", parquetPath.getFullPath(), rootPath));
+      String relativePath = parquetPath.getFullPath().replaceFirst(rootPath, "");
```

Comment:
Java defines a Path abstraction that will compute relative paths.

An [old fashioned way](http://stackoverflow.com/questions/204784/how-to-construct-a-relative-path-in-java-from-two-absolute-paths-or-urls):
```
String relative = new File(base).toURI().relativize(new File(path).toURI()).getPath();
```

From the same post, the Java 7 way:
```
        Path pathAbsolute = Paths.get("/var/data/stuff/xyz.dat");
        Path pathBase = Paths.get("/var/data");
        Path pathRelative = pathBase.relativize(pathAbsolute);
```

Hadoop's path may have something similar.